src/main/java/org/spongepowered/api/service/permission/PermissionDescription.java

```diff
@@ -103,70 +122,113 @@
     String getId();
 
     /**
-     * Gets a short description of the linked permission. May include a link to
-     * a more detailed description on the plugin's web page.
+     * Gets a short description of the linked permission.
+     *
+     * <p>May include a link to a more detailed description on the plugin's
+     * web page.</p>
+     *
+     * <p>Will return an empty optional for descriptions which have been
+     * automatically generated, or where a description was omitted when the
+     * {@link PermissionDescription} was created.</p>
      *
      * @return A short description of the linked permission
      */
-    Text getDescription();
+    Optional<Text> getDescription();
 
     /**
-     * Gets all subjects that have this permission set.
+     * Gets the owning plugin the permission belongs to.
+     *
+     * <p>Will return an empty optional for descriptions which have been
+     * automatically generated.</p>
+     *
+     * @return The owning plugin the permission belongs to
+     */
+    Optional<PluginContainer> getOwner();
```

Comment:
PermissionDescriptions can be automatically generated by the permission plugin.

As all checks are forwarded and handled by the perms plugin, they can maintain their own registry of "known" permission. (I do this in LuckPerms for tab completion, etc)

I did consider that in this case, the plugin container for the permission plugin could be used, however, the permission plugin wouldn't really be the "owner" of the permission. Also, using an Optional allows you to distinguish between PermissionDescriptions where were automatically generated, and those which were registered and "official".

The builder for PermissionDescriptions will requires a plugin instance. That remains the same.

Hopefully that logic makes sense, I'm happy to change it though if you disagree. 😄